arraycontext/impl/pytato/__init__.py

```diff
@@ -521,34 +580,64 @@ def __init__(self,
         super().__init__(compile_trace_callback=compile_trace_callback)
         self.array_types = (pt.Array, DeviceArray)
 
-    def clone(self):
-        return type(self)()
+    @property
+    def frozen_array_types(self) -> Tuple[Type, ...]:
+        from jax.numpy import DeviceArray
+        return (DeviceArray, )
+
+    def _map_container(self, func, array, allowed_types=None):
```

Comment:
`_rec_map_container`?

Reply:
Renamed in https://github.com/inducer/arraycontext/compare/f4ab671ce9eb96e9dd2397d9a77d5e7ac7c9be36..971df793aed6b5b107be09588c0791a2f664c2b9